pre-commit.nix

```diff
@@ -35,5 +35,13 @@ in nix-pre-commit-hooks.run {
      entry = "${(pythonenv.withPackages (pkgs: with pkgs; [doc8]))}/bin/doc8";
      files = "\\.rst$";
    };
+
+   mypy = {
+     enable = true;
+     name = "mypy";
+     description = "For type checking";
+     entry = "${pythonenv}/bin/mypy";
```

Comment:
This is not needed at runtime, so it would be better not to rely on it being included in `pythonenv`. We could adopt the pattern used for `doc8` here.

Reply:
This does not work because mypy needs to be in pythonenv to pick up the other libraries type definition (i.e. aiohttp and gidgethub). Also it is convenient to have mypy in nix-shell because incremental type checking is significant faster than having to run `nix-build` all the time.

Reply:
The way I've done it for `doc8` actually extends `pythoenv`, so it should work. You're probably right about having it in `nix-shell` though, `doc8` is different since its run much less frequently.